ocaml/lambda/tmc.ml

```diff
@@ -763,11 +784,12 @@ let rec choice ctx t =
         direct = (fun () -> Lapply apply_no_bailout);
       }
 
-  and choice_makeblock ctx ~tail:_ (tag, flag, shape, mode) blockargs loc =
+  and choice_makeblock ctx ~tail:_ (tag, flag, shape, is_reusing, mode) blockargs loc =
     let choices = List.map (choice ctx ~tail:false) blockargs in
     match Choice.find_nonambiguous_tmc_call choices with
     | Choice.No_tmc_call args ->
-        Choice.lambda @@ Lprim (Pmakeblock (tag, flag, shape, mode), args, loc)
+        let block = Constr.makeblock (tag, flag, shape, is_reusing, mode) in
+        Choice.lambda @@ Lprim (block, args, loc)
```

Comment:
It's not clear to me that `args` should be the same here for the two cases of `is_reusing`. That is, when `is_reusing` says to allocate, `args` must be all the fields of the constructor. But when `is_reusing` says to reuse, `args` is only the changed fields. But that's not what this code does.

Reply:
Yes, for the `tmc` analysis we only need to consider the changed fields, since only they can contain a recursive call. The same code works in both cases.